### DISCUSSION TITLE
How to debug a program that reads from stdin?

### DISCUSSION THREAD

**Discussion by gepbird** [mrcjkb/rustaceanvim#187] [Q&A] [answered]
After starting a debug session in a rust project, the DAP Console doesn't show anything from stdout and the buffer is not modifiable. The DAP REPL shows debugger logs and stdout (I'm not familliar with the DAP REPL, but maybe there's a command that can write to stdin?)

![Untitled](https://github.com/mrcjkb/rustaceanvim/assets/29818440/562fa82b-6ef9-4cca-8a4c-8263e2e33581)

I've set up DAP for python a long time ago, setting its console to integratedTerminal, and stdin and stdout works in the DAP Console.
<details><summary>debugpy config</summary>

```lua
dap.adapters.python = {
  type = 'executable',
  command = 'python',
  args = { '-m', 'debugpy.adapter' },
}

dap.configurations.python = {
  {
    type = 'python',
    name = 'launch - debugpy',
    request = 'launch',
    program = '${file}',
    console = 'integratedTerminal',
  },
}
```
</details>

**Comment by mrcjkb** [comment, marked as answer]
Hey 👋 

I don't think I can help you here, as this is an nvim-dap or lldb/codelldb issue.

Afaik, not all debug adapters support `integratedTerminal`.

Possibly related: https://github.com/mfussenegger/nvim-dap/discussions/1127
